### PR TITLE
Added Parser rules to parse UnitsML

### LINKED ISSUE #7: Parse AsciiUnits syntax and UnitsML
Integrate code from:
* https://github.com/plurimath/plurimath/pull/112

### PR DESCRIPTION
**Pull request by suleman-uzair**
Implemented the following points in the PR.

1. Added parser rules using [**Parslet gem**](https://github.com/kschiess/parslet).
2. Added [**Unitsdb**](https://github.com/unitsml/unitsdb) as submodule.

closes #7

### CODE REVIEW

File: lib/unitsml/parse.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    def arr_to_expression(arr, file_name)
+      array = arr&.flatten&.compact&.sort_by(&:length).reverse
+      array&.reduce do |expression, expr_string|
+        expression = str(expression).as(file_name.to_sym) if expression.is_a?(String)
+        expression | str(expr_string).as(file_name.to_sym)
+      end
+    end
+
+    def sqrt(rule)
+      str("sqrt") >> str("(") >> rule.as(:sqrt) >> str(")")
```
Comment (HassanAkbar) [inline lib/unitsml/parse.rb:49]:
```suggestion
      str("sqrt(") >> rule.as(:sqrt) >> str(")")
```

File: lib/unitsml/parser.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Unitsml
+  class Parser
+    attr_accessor :text
+
+    def initialize(text)
+      @regexp = %r{(quantity|name|symbol|multiplier):\s*}
+      text = text&.match(/unitsml\((.*)\)/)
+      raise "Invalid input" if text.nil?
```
Comment (HassanAkbar) [inline lib/unitsml/parser.rb:10]: I think we do not need to raise an error here. It should be possible for the user to input the text without the `unitsml()` tag.

File: lib/unitsml/unitsdb.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        unit_hash.each do |key, value|
+          value["unit_symbols"]&.each do |symbol|
+            symbols[symbol["id"]] = { id: key, fields: value } unless symbol["id"]&.empty?
+          end
+        end
+        symbols
+      end
+
+      def prefixs_ids(prefixe_hash, hash = {})
+        prefixe_hash&.map do |key, value|
```
Comment (HassanAkbar) [inline lib/unitsml/unitsdb.rb:59]: use each for looping.

**Comment by suleman-uzair** [comment]
@ronaldtse, All the checks are passed, but the last one's failing with the following error.

> **Error: Bad credentials**

**Review by ronaldtse — APPROVED** [review]

**Pull request merged**